src/js/abstract/hocs/with-render.js

```diff
@@ -0,0 +1,160 @@
+import lifecycleLogger from '../utils/lifecycle-logger';
+import { clearIsSameNode, isSameNodeOnce } from '../utils/is-same-node-once';
+import nanomorph from '../utils/component-morph';
+import TemplateNoStringReturnException from '../utils/template-no-string-return-exception';
+
+const withRender = Base =>
+  /**
+   * Adds the ability to render external DOM-based templates,
+   * applies changes incrementally by DOM-morphing and provides additional lifecycle hooks.
+   *
+   * **Light DOM**
+   * The light DOM are the provided children from the users of your component (light meaning easy to digest).
+   *
+   * ```html
+   * <axa-example>
+   *   <div>This is some light DOM for axa-example</div>
+   * </axa-example>
+   * ```
+   *
+   * **Local DOM**
+   * The local DOM is the DOM tree rendered by the component itself (in our case provided by `template`).
+   *
+   * ```js
+   * export default function(props, childrenFragment) {
+   *   return nanohtml`<article>
+   *     ${childrenFragment} <!-- light DOM injection point -->
+   *   </article>`;
+   * }
+   * ```
+   *
+   * **Flattened DOM**
+   * The flattened DOM is the final product where the user's light DOM is injected into the Components local DOM.
+   *
+   * ```html
+   * <axa-example>
+   *   <article>
+   *     <div>This is some light DOM for axa-example</div> <!-- light DOM injection point -->
+   *   </article>
+   * </axa-example>
+   * ```
+   */
+  class Render extends Base {
+    constructor(...args) {
+      super(...args);
+
+      // hook into updated
+      this.updated = this.render;
```

Comment:
@LucaMele 
@TheDadi 
`this.updated` is called by `withUpdate`.
Atm I have only this single function, though could be considered to change it to an event emitter (though no use case so far)

Reply:
then lets keep it for the moment